### ISSUE TITLE
Bulk image upload &  processing

### ISSUE THREAD

**Issue by chraebsli** [GoogleChromeLabs/squoosh#1406]
**Is your feature request related to a problem? Please describe.**
I often have to compress & resize lots of files with the same compression settings. This takes lots of time when I have to drag every image individually, click on import saved settings, wait a moment, download, repeat.

**Describe the solution you'd like**
I'd be happy if you could upload multiple files at a time, which then be resized & compressed with the same setting and download them as a ZIP archive or so

**Does other service/app have this feature?**
I regularly use [CompressPNG](https://compresspng.com/) and its other tools like [JPG2PNG](https://jpg2png.com/) for bulks, but they don't support resizing & compression settings.

**Additional context**
\-

**Comment by chraebsli** [comment]
& Thanks for the app, I really like it! I hope it gets more features soon!

**Comment by LarsFlieger** [comment]
This is definitely possible with good speeds. This websites does it: https://clevercompress.com/en


https://github.com/user-attachments/assets/ed0a92d4-7166-48c7-b83f-78bb6b1355c7



With web workers and threads you can achieve quite good speeds at several images.

**Comment by chraebsli** [comment]
Thanks for the recommendation, but that's no what I was looking for. There are a lot of smart and great compressors out there, like the one I mentioned.

But none of these have the features Squoosh has:
- custom resize and compression settings
- local processing
- see differences in real time

**Comment by LarsFlieger** [comment]
> Thanks for the recommendation, but that's no what I was looking for. There are a lot of smart and great compressors out there, like the one I mentioned.
> 
> But none of these have the features Squoosh has:
> 
> * custom resize and compression settings
> * local processing
> * see differences in real time

I understand. The tool I shared performs local processing, and I'm confident there are tools that offer local compression and settings for resizing and compression.

However, it’s true that most tools offering bulk processing lack features like difference display or similar functionalities. Perhaps we should suggest adding these features to them.

**Comment by edvinPL** [comment]
I love this tool. Recommended it to many people. It beats solutions like Caesium. Only issue is it lacks bulk processing...

**Comment by Khongchai** [comment]
I made a fork of squoosh that adds batch processing. 

https://squoosh-multiple-export.vercel.app/

Hope this is useful to some

**Comment by chraebsli** [comment]
> [squoosh-multiple-export.vercel.app](https://squoosh-multiple-export.vercel.app/)

Great work @Khongchai, Thanks! Here some feedback & bugs I discovered:
- it would be cool if the file names persisted. In my case, they just get renamed to a random UUID
- converting files of different resolutions & aspect ratios to a different filetype works flawlessly (png -> jpeg)
  - converting of multiple file types result in different sizes (same image in png + jpg -> webp). difference of about 15% of file size
  - custom resize with the same files obviously dont work
  - neither do fixed resizes like 50%, which makes all images the same size (of first file) instead of each files 50%
- in a later version, i'd like to be able to have to change the image in the preview (dropdown selector/ tabs)

---
That would be the links to the code and PR: 
- https://github.com/Khongchai/squoosh-multiple-export
- https://github.com/GoogleChromeLabs/squoosh/pull/1428

**Comment by Khongchai** [comment]
@chraebsli Thanks for the feedback! I noticed that the randomUUID problem happens only when you're using microsoft edge (I tested on Arc, Firefox and Edge, maybe chrome has this issue as well). Anyway one of my todo items is to try to zip everything in memory and download as one .gz file. 

I'll copy some of your comments to the PR. I kind of like keeping all todos in one place.

**Comment by chraebsli** [comment]
> I'll copy some of your comments to the PR. I kind of like keeping all todos in one place.

Yeah haven't seen the PR feedback at first, only searched for the link.

**Comment by Arvin-xc** [comment]
Hi there, I made some minor modifications based on squoosh to support <strong>bulk compression</strong>
<img width="1641" alt="image" src="https://github.com/user-attachments/assets/5112c7b0-9679-424f-8fb4-561ab1d474be">
You can modify the compression configuration before compressing.
Here is the App deployed on vercel: https://bulk-squoosh.vercel.app/


Hope this helps.

**Comment by ModischFabrications** [comment]
FYI: https://bulk-squoosh.vercel.app/ from @Arvin-xc doesn't work (anymore), I wasn't able to bulk compress, it seems to be exactly like the original. There is no readme and no ability to open issues to address these problems either. 

https://squoosh-multiple-export.vercel.app/ seems to work still, so better use that. It doesn't allow normalizing sizes though and has no way to report issues either.

**Comment by edvinPL** [comment]
> I made a fork of squoosh that adds batch processing.
> 
> https://squoosh-multiple-export.vercel.app/
> 
> Hope this is useful to some

This works, the only issue is the name-changing for bulk processing, any way to fix it? When downloading one-by-one the name changing doesnt happen but that's tedious for my need :)

**Comment by Khongchai** [comment]
@ModischFabrications 

You can report issues in this open PR https://github.com/GoogleChromeLabs/squoosh/pull/1428

We'll get to it...eventually

**Comment by marieliptus** [comment]
> > I made a fork of squoosh that adds batch processing.
> > https://squoosh-multiple-export.vercel.app/
> > Hope this is useful to some
> 
> This works, the only issue is the name-changing for bulk processing, any way to fix it? When downloading one-by-one the name changing doesnt happen but that's tedious for my need :)

This one works amazing,but the only problem is the name changing that it's not even in order.

**Comment by Khongchai** [comment]
Ok I fixed the file name issue just name just now -- try again please everyone

**Comment by richi-coder** [comment]
the batch squash is ok, but the download is individual, not zipped, shouldn't it?

**Comment by faizan1947** [comment]
> I made a fork of squoosh that adds batch processing.
> 
> https://squoosh-multiple-export.vercel.app/
> 
> Hope this is useful to some

Thanks , works perfect  , the file names are same as while upload

**Comment by faizan1947** [comment]
> the batch squash is ok, but the download is individual, not zipped, shouldn't it?

https://squoosh-multiple-export.vercel.app/
this one is doing zip

**Comment by sjimbonator** [comment]
> I made a fork of squoosh that adds batch processing.
> 
> https://squoosh-multiple-export.vercel.app/
> 
> Hope this is useful to some

Thank you sm!!

**Comment by nightgolfer** [comment]
I also made a fork (named **Squiish**) based on @Khongchai's version ([squoosh-multiple-export](https://github.com/Khongchai/squoosh-multiple-export)). Fixed some issues and added some new features:

- Bulk processing from [squoosh-multiple-export](https://github.com/Khongchai/squoosh-multiple-export), but added graceful fallback for situations where files would fail to download because the app ran out of memory.
- Added the ability to set the bulk batch size to mitigate errors caused by app running out of memory (Edit > Bulk processing > Batch size). Default is 3 (app will process and download 3 images at a time, before starting on the next batch of 3).
- Added the ability to proportionally resize all uploaded images based on their longest edge (Edit > Resize > Resize to > Long edge); change the 'Long edge:' value to whatever length in pixels you want each image's longest edge to have (short edge will be scaled to match). 'Long edge' is the default 'Resize to' method. Will probably add 'short edge', 'width' and 'height' in the future.
- Added the ability to rename all downloaded files (Edit > Filename > Rename). Files will have sequential numbering appended, with leading underscore and zeroes (where relevant), e.g.: new-filename_1.jpg for batches ≤ 9, new-filename_01.jpg for batches ≤ 99, etc. Default value is blank (no renaming).
- Added the ability to append text to the filename (Edit > Filename > Append), to make processed files easily identifiable. Default value is '_sqsh', e.g.: new-filename_01_sqsh.jpg. Edit > Filename is toggled on by default.

You can serve it locally, run it locally with Electron, or build a standalone Electron app. It's also on Vercel: https://squiish.vercel.app

My version doesn't do ZIP, maybe in the future. Strange, as it's based on _squoosh-multiple-export_, which comments above noted has ZIP functionality 🤔 Does ZIP really make sense, when all files are processed locally, anyway?

Throw me a star if you find it useful, I've never had one of those 😅❤️⭐

Fork code: https://github.com/nightgolfer/squiish
Live app: https://squiish.vercel.app

![Image](https://github.com/user-attachments/assets/a4bd5293-c346-4e1c-8fff-10b6be5dde9a)

**Comment by KaKi87** [comment]
Hi @Khongchai, @Arvin-xc, @nightgolfer,
Would you please be willing to enable GitHub Issues on your forks ?
Thanks
